File: src/client/src/app/modules/admin/payroll/models/salaryperks.ts
import { GlobalPerkType, SalaryPerksType } from "src/app/core/enums/SalaryPerksType";

export interface SalaryPerks {
    id: string;
    createaAt: string;
    updatedAt: string;
    organizationId: string;
    branchId: string;
    name: string;
    employeeId: string;
    type: SalaryPerksType;
    percentage: string;
    amount: string;
    isRecursion: boolean;
    isRecursionUnLimited: boolean;
    recursionEndMonth: string;
    description: string;
    effecitveFrom: string;
    isTaxable: string;
    isGlobal: string;
    globalPerkType: GlobalPerkType;
}

File: src/server/Modules/Accounting/Modules.Accounting.Core/Entities/SalaryPerks.cs
using System;
using FluentPOS.Shared.Core.Domain;
using FluentPOS.Shared.DTOs.Enums;

namespace FluentPOS.Modules.Accounting.Core.Entities
{
    public class SalaryPerks : BaseEntity
    {
        public long? EmployeeId { get; set; }

        public string Name { get; set; }

        public SalaryPerksType Type { get; set; }

        public double Percentage { get; set; }

        public double Amount { get; set; }

        public bool IsRecursion { get; set; }

        public bool IsRecursionUnLimited { get; set; }

        public DateTime? RecursionEndMonth { get; set; }

        public string Description { get; set; }

        public DateTime EffecitveFrom { get; set; }

        public bool IsTaxable { get; set; }

        public bool IsGlobal { get; set; }

        public GlobalSalaryPerksType GlobalPerkType { get; set; }
    }
}